packages/flutter_tools/lib/src/devfs.dart

```diff
@@ -268,7 +268,7 @@ class _DevFSHttpWriter {
       request.headers.removeAll(HttpHeaders.ACCEPT_ENCODING);
       request.headers.add('dev_fs_name', fsName);
       request.headers.add('dev_fs_path_b64',
-                          BASE64.encode(UTF8.encode(devicePath)));
+                          BASE64.encode(UTF8.encode(devicePath.path)));
```

Comment:
Unfortunately, the receiver of this call uses `Uri.file` [1] to turn this back into an Uri. `Uri.file` only wants the path component of an Uri.

[1] https://github.com/dart-lang/sdk/blob/66b5248bf937ef1202f04b106c63730e57a008b8/sdk/lib/vmservice/devfs.dart#L40